### DISCUSSION TITLE
Issue with reading local HF pytorch checkpoint

### DISCUSSION THREAD

**Discussion by dokooh** [obss/sahi#1024] [Q&A] [answered]
Hi, 

I would like to try testing your sliced inference on large images that my model can't detect using Sahi. 

I use a finetuned local checkpoint of a HF facebook/detr-resnet-50 model. 
This is the line I use in my code: 

```
self.default_checkpoint = 'pytorch_model.bin'

self.model = AutoDetectionModel.from_pretrained(
          model_type='huggingface',
          model_path=self.default_checkpoint,
          config_path=self.config.config_path,
          confidence_threshold=self.conf_threshold,
          image_size=640,
          device=self.DEVICE
      )
```
which leads to this error:

```
OSError: pytorch_model.bin is not a local folder and is not a valid model identifier listed on 'https://huggingface.co/models' 
If this is a private repository, make sure to pass a token having permission to this repo either by logging in with `huggingface-cli login` or by passing `token=<your_token>`
```
The version of sahi I have locally installed is: 
_sahi==0.11.15
transformers==4.36.0.dev0
torch==1.13.1_

**Comment by fcakyon** [comment, marked as answer]
Hey @dokooh for huggingface+sahi you need to provide local hf model folder to model_path instead of the bin file path.

Check the notebook for more detail: https://nbviewer.org/github/obss/sahi/blob/main/demo/inference_for_huggingface.ipynb
